### ISSUE TITLE
Bug: Error Checking / Handling on Meetup urlname value

### ISSUE THREAD

**Issue by allella** [hackgvl/hackgreenville-com#213]
@ramonaspence 

I got a fail message sort of out of the blue this evening on the events API cron job.

>   File "/home/eventapi/upstate_tech_cal_service/update_functions.py", line 65, in format_meetup_events
>     group_item = [i for i in group_list if i.get('field_events_api_key') == str(event['group']['urlname'])][0]
> IndexError: list index out of range
> 

It appears the issue is the Meetup [group][urlname] value for _DefCon864_ was all of a sudden lowercase in Meetup's API, whereas it's been CamelCase in the orgs API for some time. 

I saw this group chatting on HackGreenville about upcoming events, so it's possible / likely they just added new events and something in the process changed their "urlname".

Meetup's API seems to be case-insensitive, since the [CamelCase API call still works too like the lower case one](https://api.meetup.com/DefCon864/events?&sign=true&photo-host=public&no_earlier_than=2022-06-01T08:00:00.000&status=upcoming,cancelled,past&page=50).

However, as soon as saved the lowercase urlname in the Orgs API the cron job runs fine again.
 
I was wondering if we could add some error checking on that section and perhaps convert both values to lowercase AND output the group->name to the logger if there's a problem detected, like if there's no matching index.

I'm not sure if this is the best way to do it, but I temporarily added the following to the code to debug it.

Near the top of _update_functions.py_

```
import logging
from logging.config import fileConfig

import pprint
from pprint import pformat

fileConfig('logging_config.ini')
logger = logging.getLogger()
```

and then within the _format_meetup_events_ function after the `for event in events_raw`, add a debugging line

`        logger.error(pformat(event.get('group').get('urlname')));
`

**Comment by allella** [comment]
This issue seems like it could be impacting the new Events API too, which makes sense because the previous issue was that the Meetup API call would stop responding to request for a specific organization unless the _urlname_ used by Meetup was exactly the same as the "API Key" that we save in the Orgs API.

Previously, it seemed to either use CamelCase on some Meetup groups, or else they recently started forcing the urlname to all lowecase in the legacy REST API.

![image](https://github.com/hackgvl/hackgreenville-com/assets/1777776/d6b2da3a-6cdb-4c87-ae1a-7f0700f3c047)

Changing the "API Key" we save in the orgs API has fixed this for many other groups, but they didn't all break at once, so it must be when the organizer does something on the Meetup side that it's forcing the urlname they use internally to stop working on the old value and start needing to use the lowercase version.

**Comment by allella** [comment]
As an example, querying Meetup's REST API for the Greenville Wordpress group with **Greenville-South-Carolina-WordPress-Group** in the request still works.  For many other active orgs, we've had to change their group "_urlname_" to all lowercase.

`https://api.meetup.com/Greenville-South-Carolina-WordPress-Group/events?&sign=true&photo-host=public `

I see now that requesting the lowercase variation also works
https://api.meetup.com/greenville-south-carolina-wordpress-group/events?&sign=true&photo-host=public

So, we may be able to "fix" the issue by changing any [remaining "API Key" values in our Orgs API](https://data.openupstate.org/rest/organizations?_format=json&event_service=meetup&org_status=active) to lowercase in anticipation of them all, eventually, converting to lowercase on Meetup's side.

**Examples**

![image](https://github.com/hackgvl/hackgreenville-com/assets/1777776/9dcad33b-1d6e-44a3-a134-fed4559bf8da)

This works, but for many other orgs using the "old" urlname in the REST API request will break unless the request uses a lowercase version of the urlname value.
![image](https://github.com/hackgvl/hackgreenville-com/assets/1777776/42df0810-a244-43ac-9cba-4ca804b46deb)

**Comment by allella** [comment]
I went ahead and updated the orgs API "field_event_api_key" values to match the lowercase version of the _urlname_ values to for each group / org in the [Meetup v2 REST API](https://github.com/hackgvl/hackgreenville-com/blob/develop/EVENTS_API.md#contributor-notes).

The all lowercase urlname values work in the Meetup REST API request even if the _urlname_ itself in the response is still using a legacy / CamelCase value.

It seems all but 4 of the groups / orgs we track had changed their urlname in the last year or so, each time creating a breaking change.  So, hopefully this puts this issue to rest.

The 4 groups which remained with the non-lowercase values, which have been changed to lowercase, were all older groups on Meetup, as below.

- Greenville SC Wordpress Group
- GSP Developers Guild
- Spartanburg Wordpress Group
- Women Who Code
-
